src/main/java/org/truffleruby/stdlib/CoverageManager.java

```diff
@@ -77,8 +103,11 @@ protected void onEnter(VirtualFrame frame) {
                         if (!configured) {
                             CompilerDirectives.transferToInterpreterAndInvalidate();
                             final SourceSection sourceSection = eventContext.getInstrumentedSourceSection();
-                            lineNumber = sourceSection.getStartLine() - 1;
-                            counters = getCounters(sourceSection.getSource());
+
+                            if (getLineHasCode(sourceSection.getSource(), sourceSection.getStartLine())) {
+                                lineNumber = sourceSection.getStartLine() - 1;
+                                counters = getCounters(sourceSection.getSource());
+                            }
```

Comment:
You lost the newline separating the `configured = true` here. Extremely minor.

Reply:
Fixed, indeed it was not intentional.
I had to copy/paste from the old commit as it didn't apply cleanly.